acq4/devices/FilterWheel/FilterWheel.py

```diff
@@ -0,0 +1,402 @@
+            if self.dev.currentFWPosition != requiredPos:
+                print 'current pos:', self.dev.currentFWPosition
+                print 'requir. pos:', requiredPos
+                self.dev.setPosition(requiredPos)
+            
+    def start(self):
+        # self.shutterOpened = self.dev.getShutter()
+        # if not self.shutterOpened:
+        #     self.dev.openShutter()
+        #     time.sleep(2.0)  ## opening the shutter causes momentary power drop; give laser time to recover
+        #                      ## Note: It is recommended to keep the laser's shutter open rather than
+        #                      ## rely on this to open it for you.
+        #print state['filterCombo']
+        #print 'start filterwheel task'
+        #LaserTask.start(self)
+        pass
+        
+    def stop(self, abort):
+        pass
+        #print 'end filterwheel task'
+        #if not self.shutterOpened:
+        #    self.dev.closeShutter()
+        #LaserTask.stop(self, abort)
+
+    def isDone(self):
+        return True
+    
+class FilterWheelTaskGui(TaskGui):
+    
+    #sigSequenceChanged = QtCore.Signal(object)  ## defined upstream
+    
+    def __init__(self, dev, taskRunner):
+        
+        TaskGui.__init__(self, dev, taskRunner)
+
+        self.ui = Ui_Form()
+        self.ui.setupUi(self)
+        self.dev = dev
+        
+        filters = self.dev.listFilters()
+        filterList = self.generatFilterList(filters)
+        for i in range(len(filterList)):
+            item = self.ui.filterCombo.addItem('%s' % filterList[i][1]) 
+        
+        item = self.ui.sequenceCombo.addItem('off')
+        item = self.ui.sequenceCombo.addItem('list')
+        self.ui.sequenceListEdit.hide()
+        
+        self.ui.sequenceCombo.currentIndexChanged.connect(self.sequenceChanged)
+        self.ui.sequenceListEdit.editingFinished.connect(self.sequenceChanged)
+            
+        ## Create state group for saving/restoring state
+        self.stateGroup = pg.WidgetGroup([
+            (self.ui.filterCombo,),
+            (self.ui.sequenceCombo,),
+            (self.ui.sequenceListEdit,),
+        ])
+        
+        
+        
+    def generateTask(self, params=None):
+        print 'params in gernateTask :', params
+        state = self.stateGroup.state()
+        
+        if params is None or 'filterWheelPosition' not in params:
+            target = state['filterCombo']
+        else:
+            target = self.filterTaskList[params['filterWheelPosition']]
+        
+        task = {}
+        task['recordState'] = True
+        task['filterWheelPosition'] = target #state['filterCombo']
+        return task    
+    
+    def saveState(self, saveItems=False):
+        state = self.stateGroup.state()
+        return state
+    
+    def restoreState(self, state):
+        self.stateGroup.setState(state)
+        if state['sequenceCombo'] == 'off':
```

Comment:
This needs to hide and show as well, right? 

    self.ui.sequenceListEdit.setVisible(state['sequenceCombo'] != 'off')